src/utils/swapErrorToUserReadableMessage.tsx

```diff
@@ -22,6 +22,19 @@ export function swapErrorToUserReadableMessage(error: any): string {
     error = error.error ?? error.data?.originalError
   }
 
+  // The 4001 error code doesn't capture the case where users reject a transaction for all wallets,
+  // so we need to parse the reason for these special cases:
+  if (
+    // For rainbow wallet:
+    (reason?.match(/request/i) && reason?.match(/reject/i)) ||
+    // For frame wallet: https://frame.sh/
```

Comment:
nit: no need to include the url, you don't do so for any other wallet
nit: capitalize the wallet names (SafePal has two caps)
nit: omit or include "wallet" for every named wallet

none of these are blocking at all